solidity_contracts/src/CairoPrecompiles/PragmaCaller.sol

```diff
@@ -71,4 +130,111 @@ contract PragmaCaller {
         }
         return response;
     }
+
+    function calculateMean(PragmaCalculateMeanRequest memory request)
+        public
+        view
+        returns (PragmaSummaryStatsResponse memory response)
+    {
+        // Serialize the data request into a format compatible with the expected Pragma inputs
+        uint256[] memory data = new uint256[](request.dataType == DataType.FuturesEntry ? 6 : 5);
+        data[0] = uint256(request.dataType);
+        data[1] = request.pairId;
+        if (request.dataType == DataType.FuturesEntry) {
+            data[2] = request.expirationTimestamp;
+            data[3] = request.startTimestamp;
+            data[4] = request.endTimestamp;
+            data[5] = uint256(request.aggregationMode);
+        } else {
+            data[2] = uint256(request.startTimestamp);
+            data[3] = uint256(request.endTimestamp);
+            data[4] = uint256(request.aggregationMode);
+        }
+
+        bytes memory returnData = pragmaSummaryStats.staticcallCairo(FUNCTION_SELECTOR_CALCULATE_MEAN, data);
+
+        assembly {
+            // Load the values from the return data
+            // returnData[0:32] is the length of the return data
```

Comment:
should the return data length be checked to ensure it is a an array of 2 felts ?

Reply:
Added